pkg/ensurance/collector/nodelocal/memory.go

```diff
@@ -0,0 +1,61 @@
+package nodelocal
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/shirou/gopsutil/mem"
+	"k8s.io/klog/v2"
+
+	"github.com/gocrane/crane/pkg/common"
+	"github.com/gocrane/crane/pkg/ensurance/collector/types"
+)
+
+const (
+	memoryCollectorName = "memory"
+)
+
+func init() {
+	registerMetrics(memoryCollectorName, []types.MetricName{types.MetricNameMemoryTotalUsage, types.MetricNameMemoryTotalUtilization}, NewMemoryCollector)
+}
+
+type MemoryCollector struct {
+	data map[string][]common.TimeSeries
+}
+
+// NewMemoryCollector returns a new Collector exposing kernel/system statistics.
+func NewMemoryCollector(_ *NodeLocalContext) (nodeLocalCollector, error) {
+
+	klog.V(2).Infof("NewMemoryCollector")
+
+	var data = make(map[string][]common.TimeSeries)
+
+	return &MemoryCollector{data: data}, nil
+}
+
+func (c *MemoryCollector) collect() (map[string][]common.TimeSeries, error) {
+	var now = time.Now()
+	stat, err := mem.VirtualMemory()
+	if err != nil {
+		return map[string][]common.TimeSeries{}, err
+	}
+
+	if stat == nil {
+		return map[string][]common.TimeSeries{}, fmt.Errorf("stat is nil")
```

Comment:
return nil on error, here and everywhere

Reply:
done~